gentleman/src/components/productListing/productListing.jsx

```diff
@@ -1,49 +1,118 @@
-import {BsHeartFill} from "react-icons/bs";
-import {AiOutlineStar} from "react-icons/ai";
-import {Link } from "react-router-dom";
+import { BsHeartFill } from "react-icons/bs";
+import { AiOutlineStar, AiOutlineHeart } from "react-icons/ai";
+import { Link, useNavigate } from "react-router-dom";
 import { useProductContext } from "../../context/product-context";
 import { useUserContext } from "../../context/user-context";
-const ProductListing=()=>{
-  const {filterData}=useProductContext();
-  const {userState,userDispatch}=useUserContext();
-return <div className="product-content">
-{
-filterData.map((item)=>{
-  let a = "";let b="";
+import { useState } from "react";
+const ProductListing = () => {
+  const { filterData } = useProductContext();
+  const { userState, userDispatch, addToWishlistHandler, addToCartHandler } =
+    useUserContext();
+  const navigate = useNavigate();
+  return (
+    <div className="product-content">
+      {filterData.map((item) => {
+        let a = "";
+        let b = "";
```

Comment:
Avoid using temporary variables !!

Reply:
used it to keep a  check on wether a  specific item is getting repeated or not